ui/apiserver/apiserver.go

```diff
@@ -80,6 +114,117 @@ func (api *ApiServer) chatHandler(c *gin.Context) {
+
+	prJson, err := json.Marshal(prData)
+	if err != nil {
+		api.logger.Errorf("Error encoding JSON: %v", err)
+		c.JSON(http.StatusInternalServerError, gin.H{"error": err})
+		return
+	}
+
+	url := fmt.Sprintf("%s/pr/skill", InstructLabBotUrl)
+	resp, err := api.sendPostRequest(url, bytes.NewBuffer(prJson))
+	if err != nil {
+		api.logger.Errorf("Error sending post request to bot http server: %v -- %v", err, resp)
+		c.JSON(http.StatusInternalServerError, gin.H{"error": err})
+		return
+	}
+	defer resp.Body.Close()
+
+	responseBody := new(bytes.Buffer)
+	_, err = responseBody.ReadFrom(resp.Body)
+	if err != nil {
+		api.logger.Errorf("Error reading response body: %v", err)
+		c.JSON(http.StatusInternalServerError, gin.H{"error": err})
+	}
+
+	if resp.StatusCode != http.StatusOK {
+		api.logger.Errorf("Error response (code : %s) from bot http server: %v", resp.StatusCode, responseBody.String())
+		c.JSON(http.StatusInternalServerError, gin.H{"error": responseBody.String()})
+		return
+	}
+
+	api.logger.Infof("Skill pull request response: %v", responseBody.String())
+	c.JSON(http.StatusOK, gin.H{"msg": responseBody.String()})
+}
+
+func (api *ApiServer) knowledgePRHandler(c *gin.Context) {
+
+	var prData KnowledgePRRequest
+	if err := c.ShouldBindJSON(&prData); err != nil {
+		api.logger.Error("Failed to bind JSON:", err)
+		c.JSON(http.StatusBadRequest, gin.H{"error": "Invalid request"})
+		return
+	}
+
+	api.logger.Infof("Received Knowledge pull request data: %v", prData)
+
+	prJson, err := json.Marshal(prData)
+	if err != nil {
+		api.logger.Errorf("Error encoding JSON: %v", err)
+		c.JSON(http.StatusInternalServerError, gin.H{"error": err})
+		return
+	}
+	url := fmt.Sprintf("%s/pr/knowledge", InstructLabBotUrl)
+	resp, err := api.sendPostRequest(url, bytes.NewBuffer(prJson))
+	if err != nil {
+		api.logger.Errorf("Error sending post request to bot http server: %v -- %v", err, resp)
+		c.JSON(http.StatusInternalServerError, gin.H{"error": err})
+		return
+	}
+
+	responseBody := new(bytes.Buffer)
+	_, err = responseBody.ReadFrom(resp.Body)
+	if err != nil {
+		api.logger.Errorf("Error reading response body: %v", err)
+		c.JSON(http.StatusInternalServerError, gin.H{"error": err})
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		api.logger.Errorf("Error response (code : %s) from bot http server: %v", resp.StatusCode, responseBody.String())
+		c.JSON(http.StatusInternalServerError, gin.H{"error": responseBody.String()})
+		return
+	}
+
+	api.logger.Infof("Knowledge pull request response: %v", responseBody.String())
+
+	c.JSON(http.StatusOK, gin.H{"msg": responseBody.String()})
+}
+
+// Sent http post request using custom client with zero timeout
+func (api *ApiServer) sendPostRequest(url string, body io.Reader) (*http.Response, error) {
+	client := &http.Client{
+		Timeout: 0 * time.Second,
+		Transport: &http.Transport{
+			TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
```

Comment:
Should we add timeout options, if so I can track those down again.

Reply:
yeah, ideally we should. But want to keep it disabled till we do some more testing of it and avoid timeout related flacks during the testing.  In the backend "git clone" and retries are bit painful and can be indeterministic.